pkg/ec2provider/ec2provider.go

```diff
@@ -60,7 +66,7 @@ type ec2ProviderImpl struct {
 	instanceIdsChannel chan string
 }
 
-func New(roleARN, region string, qps int, burst int) EC2Provider {
+func New(roleARN, sourceARN, region string, qps int, burst int) EC2Provider {
```

Comment:
This new field should be optional, can we make it optional?

Reply:
Yeah it is already optional, by default it will be empty string if the value is not set in the config. https://github.com/kubernetes-sigs/aws-iam-authenticator/pull/749/files#diff-8098996ac6450091ed1b73d6860e512c51ed35c8e3fdf67dd2d0f100e1968f7bR299 - it will ignore this header if the value is empty.